# Final Project -> Waste

## Q7: What is the relationship between the total amount of waste produced, GDP and population globally? 

### Import Packages

In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

### Load Dataset

In [9]:
c_sw = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/country_level_data_0.csv")
c_sw.head()

iso3c region_id country_name income_id           gdp  \
0   ABW       LCN        Aruba       HIC           NaN   
1   AFG       SAS  Afghanistan       LIC  2.141361e+10   
2   AGO       SSF       Angola       LMC  1.030423e+11   
3   ALB       ECS      Albania       UMC  1.347108e+10   
4   AND       ECS      Andorra       HIC  3.319880e+09   

   composition_food_organic_waste_percent  composition_glass_percent  \
0                                     NaN                        NaN   
1                                     NaN                        NaN   
2                                    51.8                        6.7   
3                                    51.4                        4.5   
4                                    31.2                        8.2   

   composition_metal_percent  composition_other_percent  \
0                        NaN                        NaN   
1                        NaN                        NaN   
2                        4.4                      11.50   
3                        4.8                      15.21   
4                        2.6                      11.60   

   composition_paper_cardboard_percent  ...  \
0                                  NaN  ...   
1                                  NaN  ...   
2                                 11.9  ...   
3                                  9.9  ...   
4                                 35.1  ...   

   waste_treatment_controlled_landfill_percent  \
0                                          NaN   
1                                          NaN   
2                                          NaN   
3                                          NaN   
4                                          NaN   

   waste_treatment_incineration_percent  \
0                                   NaN   
1                                   NaN   
2                                   NaN   
3                                   NaN   
4                                  52.1   

   waste_treatment_landfill_unspecified_percent  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                                           NaN   

   waste_treatment_open_dump_percent waste_treatment_other_percent  \
0                                NaN                           NaN   
1                                NaN                           NaN   
2                                NaN                           NaN   
3                                NaN                           NaN   
4                                NaN                           NaN   

  waste_treatment_recycling_percent  \
0                              11.0   
1                               NaN   
2                               NaN   
3                               NaN   
4                               NaN   

  waste_treatment_sanitary_landfill_landfill_gas_system_percent  \
0                                                NaN              
1                                                NaN              
2                                                NaN              
3                                                NaN              
4                                                NaN              

  waste_treatment_unaccounted_for_percent  \
0                                    89.0   
1                                     NaN   
2                                     NaN   
3                                     NaN   
4                                    47.9   

  waste_treatment_waterways_marine_percent  where_where_is_this_data_measured  
0                                      NaN                                NaN  
1                                      NaN                              Other  
2                                      NaN                                NaN  
3                                      NaN                Some di

### DW: Subsetting specific columns 

In [14]:
c_sw1 = c_sw[["country_name", "gdp", "population_population_number_of_people", "total_msw_total_msw_generated_tons_year"]]
c_sw1.head()

country_name           gdp  population_population_number_of_people  \
0        Aruba           NaN                                103187.0   
1  Afghanistan  2.141361e+10                              34656032.0   
2       Angola  1.030423e+11                              25096150.0   
3      Albania  1.347108e+10                               2880703.0   
4      Andorra  3.319880e+09                                 82431.0   

   total_msw_total_msw_generated_tons_year  
0                             8.813202e+04  
1                             5.628525e+06  
2                             4.213644e+06  
3                             1.142964e+06  
4                             4.300000e+04

### Analysis: Running Correlation for Q7

In [15]:
c_sw1.corr(method='pearson').style.format("{:.2}").background_gradient(cmap=plt.get_cmap('coolwarm'), axis=1)

## Results: 

Scale: Low to High correlation ranging between 0(-#) and 1 

- Correaltion between gdp and total generated waster per year in tons is 89% -> extremely high 

- Correaltion between population and total generated waster per year in tons is 83% -> high

- Correaltion between population and gdp is 57% -> slightly above high correaltion 

_______

Credit: Devin's starter code 

# Q: 3-6

## Import Packages 

In [12]:
import pandas as pd
pd.set_option('display.max_rows', 500)

# Load Data -> England -> Waste Gen & Treatment

iloc -> index location (removing the first column - column 0)

In [13]:
england_industry_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_England_Industry.1.csv").iloc[:,1:]
england_managment_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_England_Management.1.csv").iloc[:,1:]

### Calling the column names of the datasets

In [14]:
england_industry_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation'],
      dtype='object')

In [15]:
england_managment_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies',
       'Total waste sent to final treatment'],
      dtype='object')

## Data Wrangle

#### Spliting Total and Haz/Non-Haz into two different dataframes of Industries Waste Gen

loc -> location 

calling the dataset.loc (in that df we only want where H/NH split == and != Total)

In [16]:
england_industry_total = england_industry_df.loc[england_industry_df['Hazardous/Non-hazardous split'] == "Total"]
england_industry_type = england_industry_df.loc[england_industry_df['Hazardous/Non-hazardous split'] != "Total"]

#### Spliting Total and Haz/Non-Haz into two different dataframes of Management Waste Treated 

In [17]:
england_managment_total = england_managment_df.loc[england_managment_df['Hazardous/Non-hazardous split'] == "Total"]
england_managment_type = england_managment_df.loc[england_managment_df['Hazardous/Non-hazardous split'] != "Total"]

### *England Total -> Indusrty & Management* 

In [26]:
england_managment_total

Year     EWC-STAT code  \
2    2010              01.1   
5    2010              01.2   
8    2010              01.3   
11   2010    01.4, 02, 03.1   
14   2010              03.2   
17   2010              03.3   
20   2010                05   
23   2010              06.1   
26   2010              06.2   
29   2010              06.3   
32   2010              07.1   
35   2010              07.2   
38   2010              07.3   
41   2010              07.4   
44   2010              07.5   
47   2010              07.6   
50   2010              07.7   
53   2010               08*   
56   2010              08.1   
59   2010             08.41   
62   2010              09.1   
65   2010              09.2   
68   2010              09.3   
71   2010              10.1   
74   2010              10.2   
77   2010              10.3   
80   2010                11   
83   2010              12.1   
86   2010  12.2, 12.3, 12.5   
89   2010              12.4   
92   2010              12.6   
95   2010              12.7   
98   2010          12.8, 13   
101  2010             Total   
104  2012              01.1   
107  2012              01.2   
110  2012              01.3   
113  2012    01.4, 02, 03.1   
116  2012              03.2   
119  2012              03.3   
122  2012                05   
125  2012              06.1   
128  2012              06.2   
131  2012              06.3   
134  2012              07.1   
137  2012              07.2   
140  2012              07.3   
143  2012              07.4   
146  2012              07.5   
149  2012              07.6   
152  2012              07.7   
155  2012               08*   
158  2012              08.1   
161  2012             08.41   
164  2012              09.1   
167  2012              09.2   
170  2012              09.3   
173  2012              10.1   
176  2012              10.2   
179  2012              10.3   
182  2012                11   
185  2012              12.1   
188  2012  12.2, 12.3, 12.5   
191  2012              12.4   
194  2012              12.6   
197  2012              12.7   
200  2012          12.8, 13   
203  2012             Total   
206  2014              01.1   
209  2014              01.2   
212  2014              01.3   
215  2014    01.4, 02, 03.1   
218  2014              03.2   
221  2014              03.3   
224  2014                05   
227  2014              06.1   
230  2014              06.2   
233  2014              06.3   
236  2014              07.1   
239  2014              07.2   
242  2014              07.3   
245  2014              07.4   
248  2014              07.5   
251  2014              07.6   
254  2014              07.7   
257  2014               08*   
260  2014              08.1   
263  2014             08.41   
266  2014              09.1   
269  2014              09.2   
272  2014              09.3   
275  2014              10.1   
278  2014              10.2   
281  2014              10.3   
284  2014                11   
287  2014              12.1   
290  2014  12.2, 12.3, 12.5   
293  2014              12.4   
296  2014              12.6   
299  2014              12.7   
302  2014          12.8, 13   
305  2014             Total   
308  2016              01.1   
311  2016              01.2   
314  2016              01.3   
317  2016    01.4, 02, 03.1   
320  2016              03.2   
323  2016              03.3   
326  2016                05   
329  2016              06.1   
332  2016              06.2   
335  2016              06.3   
338  2016              07.1   
341  2016              07.2   
344  2016              07.3   
347  2016              07.4   
350  2016              07.5   
353  2016              07.6   
356  2016              07.7   
359  2016               08*   
362  2016              08.1   
365  2016             08.41   
368  2016              09.1   
371  2016              09.2   
374  2016              09.3   
377  2016              10.1   
380  2016              10.2   
383  2016              10.3   
386  2

#### Combining two subject indentifier -> Year and EWC-STAT description -> which will make each row unique

- They did not match on the rows that were aggregates "Total waste generated" and "Total waste managed"

- So we removed those because they were rows that represented total's

In [32]:
combine_yr_ewc = england_industry_total[['Year', 'EWC-STAT description']] == england_managment_total[['Year', 'EWC-STAT description']]

In [22]:
combine_yr_ewc

Year  EWC-STAT description
2    True                  True
5    True                  True
8    True                  True
11   True                  True
14   True                  True
17   True                  True
20   True                  True
23   True                  True
26   True                  True
29   True                  True
32   True                  True
35   True                  True
38   True                  True
41   True                  True
44   True                  True
47   True                  True
50   True                  True
53   True                  True
56   True                  True
59   True                  True
62   True                  True
65   True                  True
68   True                  True
71   True                  True
74   True                  True
77   True                  True
80   True                  True
83   True                  True
86   True                  True
89   True                  True
92   True                  True
95   True                  True
98   True                  True
101  True                 False
104  True                  True
107  True                  True
110  True                  True
113  True                  True
116  True                  True
119  True                  True
122  True                  True
125  True                  True
128  True                  True
131  True                  True
134  True                  True
137  True                  True
140  True                  True
143  True                  True
146  True                  True
149  True                  True
152  True                  True
155  True                  True
158  True                  True
161  True                  True
164  True                  True
167  True                  True
170  True                  True
173  True                  True
176  True                  True
179  True                  True
182  True                  True
185  True                  True
188  True                  True
191  True                  True
194  True                  True
197  True                  True
200  True                  True
203  True                 False
206  True                  True
209  True                  True
212  True                  True
215  True                  True
218  True                  True
221  True                  True
224  True                  True
227  True                  True
230  True                  True
233  True                  True
236  True                  True
239  True                  True
242  True                  True
245  True                  True
248  True                  True
251  True                  True
254  True                  True
257  True                  True
260  True                  True
263  True                  True
266  True                  True
269  True                  True
272  True                  True
275  True                  True
278  True                  True
281  True                  True
284  True                  True
287  True                  True
290  True                  True
293  True                  True
296  True                  True
299  True                  True
302  True                  True
305  True                 False
308  True                  True
311  True                  True
314  True                  True
317  True                  True
320  True                  True
323  True                  True
326  True                  True
329  True                  True
332  True                  True
335  True                  True
338  True                  True
341  True                  True
344  True                  True
347  True                  True
350  True                  True
353  True                  True
356  True                  True
359  True                  True
362  True                  True
365  True                  True
368  True                  True
371  True                  True
374  

Results: the years match but the EWC-STAT do not (101, 203, 305, 407)

#### loc-> locating

- calling all the False (i.e. all the total waste from different years- ) from the column EWC to see aren't the same, so we can remove before we concatinate meaning combine two worksheets.

- if == True: it would return all rows where they do match, which would be all except 101, 203, 305, 407
- if == False: it would return all rows where they do not match, which would be 101, 203, 305, 407

In [24]:
combine_yr_ewc.loc[combine_yr_ewc['EWC-STAT description'] == False]

Year  EWC-STAT description
101  True                 False
203  True                 False
305  True                 False
407  True                 False

#### Printing the length before we remove 'Total waste generation'

In [25]:
print(len(england_industry_total))
len(england_managment_total)

136


136

#### Row Manipulation: Calling all the rows to remove the rows that have 'Total waste generation'

- Double Checking to make sure: Printing the length after remove 'Total waste generation'

In [28]:
england_industry_total = england_industry_total.loc[england_industry_total['EWC-STAT description'] != '*Total waste generation',:]
len(england_industry_total)

132

#### Row Manipulation: Calling all the rows to remove the rows that have 'Total waste treated'

- Double Checking to make sure: Printing the length after remove 'Total waste treated'

In [27]:
england_managment_total = england_managment_total.loc[england_managment_total['EWC-STAT description'] != '*Total waste treated',:]
len(england_managment_total)

132

#### Merge Worksheets

- Above lines, we made sure Year and EWC Swat were exact same before we concatenated datasets

- Now that both tables are set to True Bouleans -> meaning removed total waste gen and treated; can combine both worksheets now that the unique subject identifier are both aligned

In [29]:
total_full = pd.concat([england_industry_total, england_managment_total], axis=1)

In [30]:
len(total_full)

132

In [31]:
total_full

Year     EWC-STAT code  \
2    2010              01.1   
5    2010              01.2   
8    2010              01.3   
11   2010    01.4, 02, 03.1   
14   2010              03.2   
17   2010              03.3   
20   2010                05   
23   2010              06.1   
26   2010              06.2   
29   2010              06.3   
32   2010              07.1   
35   2010              07.2   
38   2010              07.3   
41   2010              07.4   
44   2010              07.5   
47   2010              07.6   
50   2010              07.7   
53   2010               08*   
56   2010              08.1   
59   2010             08.41   
62   2010              09.1   
65   2010              09.2   
68   2010              09.3   
71   2010              10.1   
74   2010              10.2   
77   2010              10.3   
80   2010                11   
83   2010              12.1   
86   2010  12.2, 12.3, 12.5   
89   2010              12.4   
92   2010              12.6   
95   2010              12.7   
98   2010          12.8, 13   
104  2012              01.1   
107  2012              01.2   
110  2012              01.3   
113  2012    01.4, 02, 03.1   
116  2012              03.2   
119  2012              03.3   
122  2012                05   
125  2012              06.1   
128  2012              06.2   
131  2012              06.3   
134  2012              07.1   
137  2012              07.2   
140  2012              07.3   
143  2012              07.4   
146  2012              07.5   
149  2012              07.6   
152  2012              07.7   
155  2012               08*   
158  2012              08.1   
161  2012             08.41   
164  2012              09.1   
167  2012              09.2   
170  2012              09.3   
173  2012              10.1   
176  2012              10.2   
179  2012              10.3   
182  2012                11   
185  2012              12.1   
188  2012  12.2, 12.3, 12.5   
191  2012              12.4   
194  2012              12.6   
197  2012              12.7   
200  2012          12.8, 13   
206  2014              01.1   
209  2014              01.2   
212  2014              01.3   
215  2014    01.4, 02, 03.1   
218  2014              03.2   
221  2014              03.3   
224  2014                05   
227  2014              06.1   
230  2014              06.2   
233  2014              06.3   
236  2014              07.1   
239  2014              07.2   
242  2014              07.3   
245  2014              07.4   
248  2014              07.5   
251  2014              07.6   
254  2014              07.7   
257  2014               08*   
260  2014              08.1   
263  2014             08.41   
266  2014              09.1   
269  2014              09.2   
272  2014              09.3   
275  2014              10.1   
278  2014              10.2   
281  2014              10.3   
284  2014                11   
287  2014              12.1   
290  2014  12.2, 12.3, 12.5   
293  2014              12.4   
296  2014              12.6   
299  2014              12.7   
302  2014          12.8, 13   
308  2016              01.1   
311  2016              01.2   
314  2016              01.3   
317  2016    01.4, 02, 03.1   
320  2016              03.2   
323  2016              03.3   
326  2016                05   
329  2016              06.1   
332  2016              06.2   
335  2016              06.3   
338  2016              07.1   
341  2016              07.2   
344  2016              07.3   
347  2016              07.4   
350  2016              07.5   
353  2016              07.6   
356  2016              07.7   
359  2016               08*   
362  2016              08.1   
365  2016             08.41   
368  2016              09.1   
371  2016              09.2   
374  2016              09.3   
377  2016              10.1   
380  2016              10.2   
383  2016              10.3   
386  2016                11   
389  2016              12.1   
392  2016  12.2, 12.3, 12.5   
395  2

#### *England Haz/Non-Haz -> Indusrty & Management* 

In [33]:
combine_yr_ewc2 = england_industry_type[['Year', 'EWC-STAT description']] == england_managment_type[['Year', 'EWC-STAT description']]
combine_yr_ewc2

Year  EWC-STAT description
0    True                  True
1    True                  True
3    True                  True
4    True                  True
6    True                  True
7    True                  True
9    True                  True
10   True                  True
12   True                  True
13   True                  True
15   True                  True
16   True                  True
18   True                  True
19   True                  True
21   True                  True
22   True                  True
24   True                  True
25   True                  True
27   True                  True
28   True                  True
30   True                  True
31   True                  True
33   True                  True
34   True                  True
36   True                  True
37   True                  True
39   True                  True
40   True                  True
42   True                  True
43   True                  True
45   True                  True
46   True                  True
48   True                  True
49   True                  True
51   True                  True
52   True                  True
54   True                  True
55   True                  True
57   True                  True
58   True                  True
60   True                  True
61   True                  True
63   True                  True
64   True                  True
66   True                  True
67   True                  True
69   True                  True
70   True                  True
72   True                  True
73   True                  True
75   True                  True
76   True                  True
78   True                  True
79   True                  True
81   True                  True
82   True                  True
84   True                  True
85   True                  True
87   True                  True
88   True                  True
90   True                  True
91   True                  True
93   True                  True
94   True                  True
96   True                  True
97   True                  True
99   True                 False
100  True                 False
102  True                  True
103  True                  True
105  True                  True
106  True                  True
108  True                  True
109  True                  True
111  True                  True
112  True                  True
114  True                  True
115  True                  True
117  True                  True
118  True                  True
120  True                  True
121  True                  True
123  True                  True
124  True                  True
126  True                  True
127  True                  True
129  True                  True
130  True                  True
132  True                  True
133  True                  True
135  True                  True
136  True                  True
138  True                  True
139  True                  True
141  True                  True
142  True                  True
144  True                  True
145  True                  True
147  True                  True
148  True                  True
150  True                  True
151  True                  True
153  True                  True
154  True                  True
156  True                  True
157  True                  True
159  True                  True
160  True                  True
162  True                  True
163  True                  True
165  True                  True
166  True                  True
168  True                  True
169  True                  True
171  True                  True
172  True                  True
174  True                  True
175  True                  True
177  True                  True
178  True                  True
180  True                  True
181  True                  True
183  True                  True
184  True                  True
186  

In [35]:
combine_yr_ewc2.loc[combine_yr_ewc2['EWC-STAT description'] == False]

Year  EWC-STAT description
99   True                 False
100  True                 False
201  True                 False
202  True                 False
303  True                 False
304  True                 False
405  True                 False
406  True                 False

In [36]:
print(len(england_managment_type))
len(england_industry_type)

272


272

In [42]:
england_industry_type = england_industry_type.loc[england_industry_type['EWC-STAT description'] != '*Total waste generation',:]
len(england_industry_type)

264

In [37]:
england_managment_type = england_managment_type.loc[england_managment_type['EWC-STAT description'] != '*Total waste treated',:]
len(england_managment_type)

264

In [43]:
type_full = pd.concat([england_industry_type, england_managment_type], axis=1)

In [44]:
len(type_full)

264

In [45]:
type_full

Year     EWC-STAT code  \
0    2010              01.1   
1    2010              01.1   
3    2010              01.2   
4    2010              01.2   
6    2010              01.3   
7    2010              01.3   
9    2010    01.4, 02, 03.1   
10   2010    01.4, 02, 03.1   
12   2010              03.2   
13   2010              03.2   
15   2010              03.3   
16   2010              03.3   
18   2010                05   
19   2010                05   
21   2010              06.1   
22   2010              06.1   
24   2010              06.2   
25   2010              06.2   
27   2010              06.3   
28   2010              06.3   
30   2010              07.1   
31   2010              07.1   
33   2010              07.2   
34   2010              07.2   
36   2010              07.3   
37   2010              07.3   
39   2010              07.4   
40   2010              07.4   
42   2010              07.5   
43   2010              07.5   
45   2010              07.6   
46   2010              07.6   
48   2010              07.7   
49   2010              07.7   
51   2010               08*   
52   2010               08*   
54   2010              08.1   
55   2010              08.1   
57   2010             08.41   
58   2010             08.41   
60   2010              09.1   
61   2010              09.1   
63   2010              09.2   
64   2010              09.2   
66   2010              09.3   
67   2010              09.3   
69   2010              10.1   
70   2010              10.1   
72   2010              10.2   
73   2010              10.2   
75   2010              10.3   
76   2010              10.3   
78   2010                11   
79   2010                11   
81   2010              12.1   
82   2010              12.1   
84   2010  12.2, 12.3, 12.5   
85   2010  12.2, 12.3, 12.5   
87   2010              12.4   
88   2010              12.4   
90   2010              12.6   
91   2010              12.6   
93   2010              12.7   
94   2010              12.7   
96   2010          12.8, 13   
97   2010          12.8, 13   
102  2012              01.1   
103  2012              01.1   
105  2012              01.2   
106  2012              01.2   
108  2012              01.3   
109  2012              01.3   
111  2012    01.4, 02, 03.1   
112  2012    01.4, 02, 03.1   
114  2012              03.2   
115  2012              03.2   
117  2012              03.3   
118  2012              03.3   
120  2012                05   
121  2012                05   
123  2012              06.1   
124  2012              06.1   
126  2012              06.2   
127  2012              06.2   
129  2012              06.3   
130  2012              06.3   
132  2012              07.1   
133  2012              07.1   
135  2012              07.2   
136  2012              07.2   
138  2012              07.3   
139  2012              07.3   
141  2012              07.4   
142  2012              07.4   
144  2012              07.5   
145  2012              07.5   
147  2012              07.6   
148  2012              07.6   
150  2012              07.7   
151  2012              07.7   
153  2012               08*   
154  2012               08*   
156  2012              08.1   
157  2012              08.1   
159  2012             08.41   
160  2012             08.41   
162  2012              09.1   
163  2012              09.1   
165  2012              09.2   
166  2012              09.2   
168  2012              09.3   
169  2012              09.3   
171  2012              10.1   
172  2012              10.1   
174  2012              10.2   
175  2012              10.2   
177  2012              10.3   
178  2012              10.3   
180  2012                11   
181  2012                11   
183  2012              12.1   
184  2012              12.1   
186  2012  12.2, 12.3, 12.5   
187  2012  12.2, 12.3, 12.5   
189  2012              12.4   
190  2012              12.4   
192  2012              12.6   
193  2012              12.6   
195  2

## Export CVS

- England cvs worksheets

In [46]:
type_full.to_csv('England_Type_Full.csv')
total_full.to_csv('England_Total_Full.csv')

____

# Load Data -> UK -> Waste Gen & Treatment

In [48]:
uk_industry_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_UK_Industry.1.csv").iloc[:,1:]
uk_managment_df = pd.read_csv("/Users/patelca/Desktop/WozU Data Science Program /WozU Final Project/Datasets/Data Wrangled Datasets/Updated_UK_Management.1.csv").iloc[:,1:]

In [49]:
uk_industry_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'A', 'B', 'C10-C12', 'C13-C15', 'C16',
       'C17_C18', 'C19', 'C20-C22', 'C23', 'C24_C25', 'C26-C30', 'C31-C33',
       'D', 'E36_E37_E39', 'F', 'G-U_X_G4677', 'EP_HH',
       'Total waste generation'],
      dtype='object')

In [51]:
uk_managment_df.columns

Index(['Year', 'EWC-STAT code', 'EWC-STAT description',
       'Hazardous/Non-hazardous split', 'Energy recovery', 'Incineration',
       'Recovery other than energy recovery - Except backfilling',
       'Recovery other than energy recovery - Backfilling',
       'Deposit onto or into land',
       'Land treatment and release into water bodies',
       'Total waste sent to final treatment'],
      dtype='object')

In [52]:
uk_industry_total = uk_industry_df.loc[uk_industry_df['Hazardous/Non-hazardous split'] == "Total"]
uk_industry_type = uk_industry_df.loc[uk_industry_df['Hazardous/Non-hazardous split'] != "Total"]

In [53]:
uk_managment_total = uk_managment_df.loc[uk_managment_df['Hazardous/Non-hazardous split'] == "Total"]
uk_managment_type = uk_managment_df.loc[uk_managment_df['Hazardous/Non-hazardous split'] != "Total"]

### UK Total

In [55]:
combine_yr_ewc3 = uk_industry_total[['Year', 'EWC-STAT description']] == uk_managment_total[['Year', 'EWC-STAT description']]

In [57]:
combine_yr_ewc3

Year  EWC-STAT description
2    True                  True
5    True                  True
8    True                  True
11   True                  True
14   True                  True
17   True                  True
20   True                  True
23   True                  True
26   True                  True
29   True                  True
32   True                  True
35   True                  True
38   True                  True
41   True                  True
44   True                  True
47   True                  True
50   True                  True
53   True                  True
56   True                  True
59   True                  True
62   True                  True
65   True                  True
68   True                  True
71   True                  True
74   True                  True
77   True                  True
80   True                  True
83   True                  True
86   True                  True
89   True                  True
92   True                  True
95   True                  True
98   True                  True
101  True                 False
104  True                  True
107  True                  True
110  True                  True
113  True                  True
116  True                  True
119  True                  True
122  True                  True
125  True                  True
128  True                  True
131  True                  True
134  True                  True
137  True                  True
140  True                  True
143  True                  True
146  True                  True
149  True                  True
152  True                  True
155  True                  True
158  True                  True
161  True                  True
164  True                  True
167  True                  True
170  True                  True
173  True                  True
176  True                  True
179  True                  True
182  True                  True
185  True                  True
188  True                  True
191  True                  True
194  True                  True
197  True                  True
200  True                  True
203  True                 False
206  True                  True
209  True                  True
212  True                  True
215  True                  True
218  True                  True
221  True                  True
224  True                  True
227  True                  True
230  True                  True
233  True                  True
236  True                  True
239  True                  True
242  True                  True
245  True                  True
248  True                  True
251  True                  True
254  True                  True
257  True                  True
260  True                  True
263  True                  True
266  True                  True
269  True                  True
272  True                  True
275  True                  True
278  True                  True
281  True                  True
284  True                  True
287  True                  True
290  True                  True
293  True                  True
296  True                  True
299  True                  True
302  True                  True
305  True                 False
308  True                  True
311  True                  True
314  True                  True
317  True                  True
320  True                  True
323  True                  True
326  True                  True
329  True                  True
332  True                  True
335  True                  True
338  True                  True
341  True                  True
344  True                  True
347  True                  True
350  True                  True
353  True                  True
356  True                  True
359  True                  True
362  True                  True
365  True                  True
368  True                  True
371  True                  True
374  

In [58]:
combine_yr_ewc3.loc[combine_yr_ewc3['EWC-STAT description'] == False]

Year  EWC-STAT description
101  True                 False
203  True                 False
305  True                 False
407  True                 False

In [59]:
print(len(uk_industry_total))
len(uk_managment_total)

136


136

In [60]:
uk_industry_total = uk_industry_total.loc[uk_industry_total['EWC-STAT description'] != '*Total waste generation',:]
len(uk_industry_total)

132

In [61]:
uk_managment_total = uk_managment_total.loc[uk_managment_total['EWC-STAT description'] != '*Total waste treated',:]
len(uk_managment_total)

132

In [62]:
total_full_uk = pd.concat([uk_industry_total, uk_managment_total], axis=1)

In [63]:
len(total_full_uk)

132

In [64]:
total_full_uk

Year     EWC-STAT code  \
2    2010              01.1   
5    2010              01.2   
8    2010              01.3   
11   2010    01.4, 02, 03.1   
14   2010              03.2   
17   2010              03.3   
20   2010                05   
23   2010              06.1   
26   2010              06.2   
29   2010              06.3   
32   2010              07.1   
35   2010              07.2   
38   2010              07.3   
41   2010              07.4   
44   2010              07.5   
47   2010              07.6   
50   2010              07.7   
53   2010               08*   
56   2010              08.1   
59   2010             08.41   
62   2010              09.1   
65   2010              09.2   
68   2010              09.3   
71   2010              10.1   
74   2010              10.2   
77   2010              10.3   
80   2010                11   
83   2010              12.1   
86   2010  12.2, 12.3, 12.5   
89   2010              12.4   
92   2010              12.6   
95   2010              12.7   
98   2010          12.8, 13   
104  2012              01.1   
107  2012              01.2   
110  2012              01.3   
113  2012    01.4, 02, 03.1   
116  2012              03.2   
119  2012              03.3   
122  2012                05   
125  2012              06.1   
128  2012              06.2   
131  2012              06.3   
134  2012              07.1   
137  2012              07.2   
140  2012              07.3   
143  2012              07.4   
146  2012              07.5   
149  2012              07.6   
152  2012              07.7   
155  2012               08*   
158  2012              08.1   
161  2012             08.41   
164  2012              09.1   
167  2012              09.2   
170  2012              09.3   
173  2012              10.1   
176  2012              10.2   
179  2012              10.3   
182  2012                11   
185  2012              12.1   
188  2012  12.2, 12.3, 12.5   
191  2012              12.4   
194  2012              12.6   
197  2012              12.7   
200  2012          12.8, 13   
206  2014              01.1   
209  2014              01.2   
212  2014              01.3   
215  2014    01.4, 02, 03.1   
218  2014              03.2   
221  2014              03.3   
224  2014                05   
227  2014              06.1   
230  2014              06.2   
233  2014              06.3   
236  2014              07.1   
239  2014              07.2   
242  2014              07.3   
245  2014              07.4   
248  2014              07.5   
251  2014              07.6   
254  2014              07.7   
257  2014               08*   
260  2014              08.1   
263  2014             08.41   
266  2014              09.1   
269  2014              09.2   
272  2014              09.3   
275  2014              10.1   
278  2014              10.2   
281  2014              10.3   
284  2014                11   
287  2014              12.1   
290  2014  12.2, 12.3, 12.5   
293  2014              12.4   
296  2014              12.6   
299  2014              12.7   
302  2014          12.8, 13   
308  2016              01.1   
311  2016              01.2   
314  2016              01.3   
317  2016    01.4, 02, 03.1   
320  2016              03.2   
323  2016              03.3   
326  2016                05   
329  2016              06.1   
332  2016              06.2   
335  2016              06.3   
338  2016              07.1   
341  2016              07.2   
344  2016              07.3   
347  2016              07.4   
350  2016              07.5   
353  2016              07.6   
356  2016              07.7   
359  2016               08*   
362  2016              08.1   
365  2016             08.41   
368  2016              09.1   
371  2016              09.2   
374  2016              09.3   
377  2016              10.1   
380  2016              10.2   
383  2016              10.3   
386  2016                11   
389  2016              12.1   
392  2016  12.2, 12.3, 12.5   
395  2

### UK Type - Haz/NHaz

In [66]:
combine_yr_ewc4 = uk_industry_type[['Year', 'EWC-STAT description']] == uk_managment_type[['Year', 'EWC-STAT description']]
combine_yr_ewc4

Year  EWC-STAT description
0    True                  True
1    True                  True
3    True                  True
4    True                  True
6    True                  True
7    True                  True
9    True                  True
10   True                  True
12   True                  True
13   True                  True
15   True                  True
16   True                  True
18   True                  True
19   True                  True
21   True                  True
22   True                  True
24   True                  True
25   True                  True
27   True                  True
28   True                  True
30   True                  True
31   True                  True
33   True                  True
34   True                  True
36   True                  True
37   True                  True
39   True                  True
40   True                  True
42   True                  True
43   True                  True
45   True                  True
46   True                  True
48   True                  True
49   True                  True
51   True                  True
52   True                  True
54   True                  True
55   True                  True
57   True                  True
58   True                  True
60   True                  True
61   True                  True
63   True                  True
64   True                  True
66   True                  True
67   True                  True
69   True                  True
70   True                  True
72   True                  True
73   True                  True
75   True                  True
76   True                  True
78   True                  True
79   True                  True
81   True                  True
82   True                  True
84   True                  True
85   True                  True
87   True                  True
88   True                  True
90   True                  True
91   True                  True
93   True                  True
94   True                  True
96   True                  True
97   True                  True
99   True                 False
100  True                 False
102  True                  True
103  True                  True
105  True                  True
106  True                  True
108  True                  True
109  True                  True
111  True                  True
112  True                  True
114  True                  True
115  True                  True
117  True                  True
118  True                  True
120  True                  True
121  True                  True
123  True                  True
124  True                  True
126  True                  True
127  True                  True
129  True                  True
130  True                  True
132  True                  True
133  True                  True
135  True                  True
136  True                  True
138  True                  True
139  True                  True
141  True                  True
142  True                  True
144  True                  True
145  True                  True
147  True                  True
148  True                  True
150  True                  True
151  True                  True
153  True                  True
154  True                  True
156  True                  True
157  True                  True
159  True                  True
160  True                  True
162  True                  True
163  True                  True
165  True                  True
166  True                  True
168  True                  True
169  True                  True
171  True                  True
172  True                  True
174  True                  True
175  True                  True
177  True                  True
178  True                  True
180  True                  True
181  True                  True
183  True                  True
184  True                  True
186  

In [67]:
combine_yr_ewc4.loc[combine_yr_ewc4['EWC-STAT description'] == False]

Year  EWC-STAT description
99   True                 False
100  True                 False
201  True                 False
202  True                 False
303  True                 False
304  True                 False
405  True                 False
406  True                 False

In [68]:
print(len(uk_industry_type))
len(uk_managment_type)

272


272

In [70]:
uk_industry_type = uk_industry_type.loc[uk_industry_type['EWC-STAT description'] != '*Total waste generation',:]
len(uk_industry_type)

264

In [71]:
uk_managment_type = uk_managment_type.loc[uk_managment_type['EWC-STAT description'] != '*Total waste treated',:]
len(uk_managment_type)

264

In [72]:
type_full_uk = pd.concat([uk_industry_type, uk_managment_type], axis=1)
len(type_full_uk)

264

In [73]:
type_full_uk

Year     EWC-STAT code  \
0    2010              01.1   
1    2010              01.1   
3    2010              01.2   
4    2010              01.2   
6    2010              01.3   
7    2010              01.3   
9    2010    01.4, 02, 03.1   
10   2010    01.4, 02, 03.1   
12   2010              03.2   
13   2010              03.2   
15   2010              03.3   
16   2010              03.3   
18   2010                05   
19   2010                05   
21   2010              06.1   
22   2010              06.1   
24   2010              06.2   
25   2010              06.2   
27   2010              06.3   
28   2010              06.3   
30   2010              07.1   
31   2010              07.1   
33   2010              07.2   
34   2010              07.2   
36   2010              07.3   
37   2010              07.3   
39   2010              07.4   
40   2010              07.4   
42   2010              07.5   
43   2010              07.5   
45   2010              07.6   
46   2010              07.6   
48   2010              07.7   
49   2010              07.7   
51   2010               08*   
52   2010               08*   
54   2010              08.1   
55   2010              08.1   
57   2010             08.41   
58   2010             08.41   
60   2010              09.1   
61   2010              09.1   
63   2010              09.2   
64   2010              09.2   
66   2010              09.3   
67   2010              09.3   
69   2010              10.1   
70   2010              10.1   
72   2010              10.2   
73   2010              10.2   
75   2010              10.3   
76   2010              10.3   
78   2010                11   
79   2010                11   
81   2010              12.1   
82   2010              12.1   
84   2010  12.2, 12.3, 12.5   
85   2010  12.2, 12.3, 12.5   
87   2010              12.4   
88   2010              12.4   
90   2010              12.6   
91   2010              12.6   
93   2010              12.7   
94   2010              12.7   
96   2010          12.8, 13   
97   2010          12.8, 13   
102  2012              01.1   
103  2012              01.1   
105  2012              01.2   
106  2012              01.2   
108  2012              01.3   
109  2012              01.3   
111  2012    01.4, 02, 03.1   
112  2012    01.4, 02, 03.1   
114  2012              03.2   
115  2012              03.2   
117  2012              03.3   
118  2012              03.3   
120  2012                05   
121  2012                05   
123  2012              06.1   
124  2012              06.1   
126  2012              06.2   
127  2012              06.2   
129  2012              06.3   
130  2012              06.3   
132  2012              07.1   
133  2012              07.1   
135  2012              07.2   
136  2012              07.2   
138  2012              07.3   
139  2012              07.3   
141  2012              07.4   
142  2012              07.4   
144  2012              07.5   
145  2012              07.5   
147  2012              07.6   
148  2012              07.6   
150  2012              07.7   
151  2012              07.7   
153  2012               08*   
154  2012               08*   
156  2012              08.1   
157  2012              08.1   
159  2012             08.41   
160  2012             08.41   
162  2012              09.1   
163  2012              09.1   
165  2012              09.2   
166  2012              09.2   
168  2012              09.3   
169  2012              09.3   
171  2012              10.1   
172  2012              10.1   
174  2012              10.2   
175  2012              10.2   
177  2012              10.3   
178  2012              10.3   
180  2012                11   
181  2012                11   
183  2012              12.1   
184  2012              12.1   
186  2012  12.2, 12.3, 12.5   
187  2012  12.2, 12.3, 12.5   
189  2012              12.4   
190  2012              12.4   
192  2012              12.6   
193  2012              12.6   
195  2

## Export CVS

- UK cvs worksheets

In [76]:
total_full_uk.to_csv('UK_Total_Full.csv')
type_full_uk.to_csv('UK_Type_Full.csv')

_____

# ANALYSIS